src/pydata_sphinx_theme/translator.py

```diff
@@ -23,11 +24,35 @@ def __init__(self, *args, **kwds):
         self.settings.table_style = "table"
 
     def starttag(self, *args, **kwargs):
-        """Ensure an aria-level is set for any heading role."""
+        """Perform small modifications to tags.
+
+        - ensure aria-level is set for any tag with heading role
+        - ensure <pre> tags have tabindex="0".
+        """
         if kwargs.get("ROLE") == "heading" and "ARIA-LEVEL" not in kwargs:
             kwargs["ARIA-LEVEL"] = "2"
+
+        if "pre" in args:
+            kwargs["tabindex"] = "0"
+
         return super().starttag(*args, **kwargs)
 
+    def visit_literal_block(self, node):
+        """Modify literal blocks.
+
+        - ensure tabindex="0" for <pre> tags buried in the HTML tree that Sphinx
+          (with Pygments) generates for literal blocks
+        """
+        try:
+            super().visit_literal_block(node)
+        except nodes.SkipNode:
+            # If the super method raises nodes.SkipNode, then we know it
+            # executed successfully and appended to self.body a string of HTML
+            # representing the code block, which we then modify.
+            html_string = self.body[-1]
+            self.body[-1] = html_string.replace("<pre", '<pre tabindex="0"')
```

Comment:
This is kinda dirty but... it seems to work 🤷 

I'll leave it up to the reviewers to decide if it's too dirty

Reply:
I am having a bit of trouble following here - is the goal of these to find and traverse any hidden/nested `pre`tags then add the `tabindex`?

Reply:
this doesn't seem that "dirty" to me.

Reply:
Ditto not dirty just working with what we have

Reply:
I read something since working on this PR that makes me wonder if this is the right thing to do. The thing I read was about tables but the same could be said about pre-blocks:

> Of course, we don't want to make the table container focusable unless its contents overflow. Otherwise we're adding a tab stop to the focus order which doesn't do anything. In my opinion, that would be a fail under [2.4.3 Focus Order](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html). Giving keyboard users elements to focus which don't actually do anything is confusing and obstructive.
> Source: Inclusive Components - Data Tables - Only Focusable Where Scrollable

Which makes me wonder if we should put a data attribute on the pre-tag and then check it with JavaScript after it loads to determine if it should really be a tab stop or not.

Reply:
Follow-up:

- https://github.com/pydata/pydata-sphinx-theme/pull/1777